pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -277,12 +319,74 @@ func updateApplicationLayerWithDefaults(al *operatorv1.ApplicationLayer) {
 
 // validateApplicationLayer validates ApplicationLayer
 func validateApplicationLayer(al *operatorv1.ApplicationLayer) error {
-	lcSpec := al.Spec.LogCollection
 
-	// If ApplicationLayer spec exists then LogCollection should be set.
-	// TODO: when we will have multiple features in future this should change to at least one feature being set
-	if lcSpec == nil {
-		return fmt.Errorf("missing required LogCollection spec in applicationLayer")
+	// If ApplicationLayer spec exists then one of its features should be set.
+	if al.Spec.LogCollection == nil && al.Spec.WebApplicationFirewall == nil {
+		return fmt.Errorf("at least one of webApplicationFirewall or logCollector must be specified on ApplicationLayer resource")
+	}
+
+	return nil
+}
+
+// getModSecurityRuleSet returns 'owasp-ruleset-config' ConfigMap from calico-operator namespace.
+// The ConfigMap is meant to contain rule set files for ModSecurity library.
+// If the ConfigMap does not exist a ConfigMap with OWASP provided Core Rule Set will be returned.
+// The rule set was cloned from https://github.com/coreruleset/coreruleset/
+func (r *ReconcileApplicationLayer) getModSecurityRuleSet(ctx context.Context) (*corev1.ConfigMap, bool, error) {
+	ruleset := new(corev1.ConfigMap)
+
+	if err := r.client.Get(
+		ctx,
+		types.NamespacedName{
+			Namespace: common.OperatorNamespace(),
+			Name:      applicationlayer.ModSecurityRulesetConfigMapName,
+		},
+		ruleset,
+	); err == nil {
+		return ruleset, false, nil
+	} else if !apierrors.IsNotFound(err) {
+		return nil, false, err
+	}
+
+	if ruleset, err := getDefaultCoreRuleset(ctx); err == nil {
+		return ruleset, true, nil
```

Comment:
Another nit, but a more idiomatic way to do this in golang would be to put the error condition first.


```go
ruleset, err := getDefaultCoreRuleset(ctx)
if err != nil {
  return nil, false, err
}
return ruleset, true, nil
```

Having read lots of go code, seeing the branches the way they are makes me immediately think there's something atypical going on here when actually it's just a normal error branch.

Reply:
sorted.